orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTaskSpec.groovy

```diff
@@ -146,15 +146,58 @@ class MonitorKatoTaskSpec extends Specification {
     "about to fail"      | MonitorKatoTask.TASK_NOT_FOUND_TIMEOUT     |  true         | 'N/A'
   }
 
+  @Unroll
+  def "should skip replicas when initial timeout is exceeded #desc"() {
+    given:
+    def ctx = [
+      "kato.last.task.id": new TaskId(taskId)
+    ]
+    if (previousRetry) {
+      ctx.put('kato.task.firstNotFoundRetry', now.minusMillis(elapsed).toEpochMilli())
+    } else {
+      if (previousRetryFlag != null) {
+        ctx.put('kato.task.firstNotFoundRetry', previousRetryFlag)
+      }
+    }
+
+    def stage = new Stage(Execution.newPipeline("orca"), "whatever", ctx)
+    task.kato = Stub(KatoService) {
+      lookupTask(taskId, false) >> { retrofit404() }
+      lookupTask(taskId, true) >>
+        Observable.from(new Task(taskId, new Task.Status(completed: true), [], []))
+    }
+
+    when:
+    def result = task.execute(stage)
+
+    then:
+    result.status == ExecutionStatus.RUNNING
+    result.context['kato.task.notFoundRetryCount'] == 1
+    if (!skipReplicaFlag) {
+      result.context['kato.task.firstNotFoundRetry.skipReplica'] == null
```

Comment:
I would avoid the if/else in your `then` block and just make this equal some concrete variable you specify in your `where` block down below.

ie.

```
blah | blarg || shouldSkipReplica
x    | y     || true
```